src/mgr/Action.h

```diff
@@ -76,11 +78,16 @@ class Action: public RefCountable
     /// calculate and keep local action-specific information
     virtual void collect() {}
 
+    /// write manager report output to a stream.
+    virtual void print(std::ostream &) {}
+
     /** start writing action-specific info to Store entry;
      * may collect info during dump, especially if collect() did nothing
      * non-atomic() actions may continue writing asynchronously after returning
+     *
+     * \deprecated implement collect() and print() instead
      */
-    virtual void dump(StoreEntry *) {}
+    virtual void dump(StoreEntry *);
```

Comment:
Would it make sense to make this pure virtual?

Reply:
Not really. That would require all child classes to implement this method.  The plan/goal is to **stop** them implementing it.

Reply:
I may be getting a sense of the strategy you have in mind